pennylane/drawer/utils.py

```diff
@@ -100,3 +101,78 @@ def unwrap_controls(op):
 
     control_values = [bool(int(i)) for i in control_values] if control_values else control_values
     return control_wires, control_values
+
+
+def find_mid_measure_cond_connections(operations, layers):
+    """Collect and return information about connections between mid-circuit measurements
+    and classical conditions.
+
+    This utility function returns three items needed for processing mid-circuit measurements
+    and classical conditions for drawing:
+
+    * A dictionary mapping each mid-circuit measurement to a corresponding bit index.
+        This map only contains mid-circuit measurements that are used for classical conditioning.
+    * A list where each index is a bit and the values are the indices  of the layers containing
+        the mid-circuit measurement corresponding to the bits.
+    * A list where each index is a bit and the values are the indices of the last layers that
+        use those bits for classical conditions.
+
+    Args:
+        operations (list[~.Operation]): List of operations on the tape
+        layers (list[list[~.Operation]]): List of drawable layers containing list of operations
+            for each layer
+
+    Returns:
+        tuple[dict, list, list]: Data structures needed for correctly drawing classical conditions
+        as described above.
+    """
+
+    # Map between mid-circuit measurements and their position on the drawing
+    # The bit map only contains mid-circuit measurements that are used in
+    # classical conditions.
+    bit_map = {}
```

Comment:
I did this because all of the returned items are not empty only if there are conditional operations. Otherwise I get a `reference before assignment` error.